Cython/Compiler/ModuleNode.py

```diff
@@ -177,7 +177,7 @@ def extend_if_not_in(L1, L2):
             # Ensure that we don't generate import code for these entries!
             for entry in scope.c_class_entries:
                 entry.type.module_name = self.full_module_name
-                entry.type.scope.directives["internal"] = True
+                entry.type.scope.directives["internal"] = internalise_c_class_entries
 
             self.scope.merge_in(scope)
```

Comment:
`merge_scope=True` seems to do something entirely separate. It should be a separate function. And `internalise_c_class_entries()` could also be a separate function that gets called in the appropriate place(s).

Reply:
As far as I can tell, we are only using this feature in utility code and not pxd injected code. Isn't this something that `UtilityCode.get_tree()` could do already?